client/src/files.ts

```diff
@@ -2,18 +2,22 @@ import vscode = require('vscode');
 import fg = require('fast-glob');
 import path = require('path');
 import { ExtensionData } from './constants';
+import { getExtensionConfiguration } from './utils';
 
-const excludeFromFileSearch: string[] = [
-    path.join('!**', 'node_modules')
-];
+function excludeFromFileSearch(wsPath:string): string[] {
+    return [
+        path.join('!**', 'node_modules'),
+        ...getExtensionConfiguration().excludePaths.map(excludedPath => path.join('!',wsPath,excludedPath))
```

Comment:
`excludedPath => path.join('!',wsPath,excludedPath)` will result in creating a path where `!` is considered a directory (because `path.join` will append a separator after `!`), so this negation won't work.

I think we need to go for something more like `excludedPath => '!' + path.join(wsPath, excludedPath)` here, so the path will be just prefixed with the '!', not considered a part of the path.

Reply:
Thanks for the quick feedback! I made your suggested update. In the examples I had tested the `wsPath` was absolute so it happened to work anyway, but this makes more sense.

Reply:
Yes, on unix based systems it works like that (because `wsPath` starts with a `/` by default and `path.join` will just prefix it) but windows users have their `wsPath` start like `d:/` and `path` will join it like `!/d:/...` and that wouldn't work. :)